packages/backend/src/models/NotificationsModel/NotificationsModel.ts

```diff
@@ -0,0 +1,125 @@
+    database: Knex;
+};
+
+export class NotificationsModel {
+    private readonly database: Knex;
+
+    constructor(deps: NotificationsModelDependencies) {
+        this.database = deps.database;
+    }
+
+    async getDashboardCommentNotifications(
+        userUuid: string,
+    ): Promise<Notification[]> {
+        const notificationsWithUserAndDashboardAndAuthor: (DbNotifications &
+            Pick<DbDashboardTileComments, 'dashboard_tile_uuid'> & {
+                user_first_name: string;
+                user_last_name: string;
+                dashboard_name: string;
+                author_first_name: string;
+                author_last_name: string;
+            })[] = await this.database(NotificationsTableName)
+            .join(
+                UserTableName,
+                `${NotificationsTableName}.user_uuid`,
+                '=',
+                `${UserTableName}.user_uuid`,
+            )
+            .join(
+                DashboardTileCommentsTableName,
+                `${NotificationsTableName}.comment_id`,
+                '=',
+                `${DashboardTileCommentsTableName}.comment_id`,
+            )
+            .join(
+                `${UserTableName} as comment_authors`,
+                `${DashboardTileCommentsTableName}.user_uuid`,
+                '=',
+                'comment_authors.user_uuid',
+            )
+            .join(
+                DashboardsTableName,
+                `${NotificationsTableName}.dashboard_uuid`,
+                '=',
+                `${DashboardsTableName}.dashboard_uuid`,
+            )
+            .select(
+                `${NotificationsTableName}.*`,
+                `${UserTableName}.first_name as user_first_name`,
+                `${UserTableName}.last_name as user_last_name`,
+                `${DashboardsTableName}.name as dashboard_name`,
+                `${DashboardTileCommentsTableName}.dashboard_tile_uuid`,
+                'comment_authors.first_name as author_first_name',
+                'comment_authors.last_name as author_last_name',
+            )
+            .where(`${NotificationsTableName}.user_uuid`, userUuid)
+            .orderBy(`${NotificationsTableName}.created_at`, 'desc');
+
+        return notificationsWithUserAndDashboardAndAuthor.map((notif) => ({
+            notificationId: notif.notification_id,
+            user: {
+                name: `${notif.user_first_name} ${notif.user_last_name}`,
+            },
+            author: {
+                name: `${notif.author_first_name} ${notif.author_last_name}`,
+            },
+            dashboard: {
+                uuid: notif.dashboard_uuid,
+                name: notif.dashboard_name,
+                tileUuid: notif.dashboard_tile_uuid,
+            },
+            viewed: notif.viewed,
+            createdAt: notif.created_at,
+        }));
+    }
+
+    async markNotificationAsRead(notificationUuid: string) {
+        return this.database('notifications')
```

Comment:
should we await here ? in case an error happens and it is not handled.

Reply:
apparently it doesn't let me 🤔 
<img width="1232" alt="image" src="https://github.com/lightdash/lightdash/assets/7611706/85a8765e-5cf8-4c92-a2a0-7cc6315f2e10">